datafusion/physical-expr-common/src/utils.rs

```diff
@@ -100,15 +103,37 @@ pub fn reverse_order_bys(order_bys: &[PhysicalSortExpr]) -> Vec<PhysicalSortExpr
         .collect()
 }
 
+/// Converts `datafusion_expr::Expr` into corresponding `Arc<dyn PhysicalExpr>`.
+/// If conversion is not supported yet, returns Error.
+pub fn convert_logical_expr_to_physical_expr(
```

Comment:
I think @jayzhan211  ran into this as well when trying to move these functions (e.g. that the AggregateUDFImpl is in terms of logical `Expr` but often needs physical exps).

One poential challenge with this implementation is the limited type support

We already have this API to convert logical to physical exprs (but it isn't available in the implementation of first and last)
https://docs.rs/datafusion/latest/datafusion/execution/context/struct.SessionState.html#method.create_physical_expr

So I guess it feels to me like it would be good to avoid the duplication somehow, but I don't have a great idea of how to do it. 

Maybe we could call this function (and the sort one) something like `limited_convert_logical_expr_to_physical_expr` to convey that it only does some basic ones 🤔

Reply:
renamed them